conf_sp_idp/sp/app/app.py

```diff
@@ -0,0 +1,204 @@
+# -*- coding: utf-8 -*-
+"""
+
+    conf_sp_idp.sp.app.app.py
+    ~~~~~~~~~~~~~~~~~~~~~~~~~
+
+    Service provider for ensure SSO process with pysaml2
+
+    This file is part of MSS.
+
+    :copyright: Copyright 2023 Nilupul Manodya
+    :copyright: Copyright 2016-2023 by the MSS team, see AUTHORS.
+    :license: APACHE-2.0, see LICENSE for details.
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+"""
+
+# Parts of the code
+
+import random
+import string
+
+from flask_login.utils import login_required, logout_user
+from flask_login import LoginManager, UserMixin, login_user
+
+from saml2.config import SPConfig
+from saml2.client import Saml2Client
+from saml2.metadata import create_metadata_string
+from saml2 import BINDING_HTTP_REDIRECT, BINDING_HTTP_POST
+from flask import Flask, redirect, request, render_template, url_for
+from flask.wrappers import Response
+import yaml
+from yaml.loader import SafeLoader
+
+from flask_sqlalchemy import SQLAlchemy
+from flask_migrate import Migrate
+
+
+
+app = Flask(__name__)
+app.config["SQLALCHEMY_DATABASE_URI"] = "sqlite:///db.sqlite"
+app.config[
+    "SECRET_KEY"
+] = "192b9bdd22ab9ed4d12e236c78afcb9a393ec15f71bbf5dc987d54727823bcbf"
```

Comment:
We should keep in mind that such secret keys should not end up in git as soon as this is for more than just testing purposes.